ValheimPlus/Configurations/ConfigurationExtra.cs

```diff
@@ -75,28 +75,29 @@ public static Configuration LoadFromIni(string filename)
             return conf;
         }
 
-        public static Configuration LoadFromIni(Stream iniStream)
+        public static void LoadFromIniToConfiguration(Stream iniStream, Configuration config)
         {
             using (StreamReader iniReader = new StreamReader(iniStream))
             {
                 FileIniDataParser parser = new FileIniDataParser();
                 IniData configdata = parser.ReadData(iniReader);
 
-                Configuration conf = new Configuration();
                 foreach (var prop in typeof(Configuration).GetProperties())
                 {
                     string keyName = prop.Name;
-                    MethodInfo method = prop.PropertyType.GetMethod("LoadIni",
-                        BindingFlags.Public | BindingFlags.Static | BindingFlags.FlattenHierarchy);
-
-                    if (method != null)
+                    MethodInfo method = prop.PropertyType.GetMethod("LoadIni", BindingFlags.Public | BindingFlags.Static | BindingFlags.FlattenHierarchy);
+                    // Only try to load syncConfig (which are the only sent), it will avoid some local configuration revert to default values
+                    if (configdata[keyName] != null && configdata[keyName]["enabled"] != null)
                     {
-                        object result = method.Invoke(null, new object[] {configdata, keyName});
-                        prop.SetValue(conf, result, null);
+                        if (method != null)
+                        {
+                            object result = method.Invoke(null, new object[] { configdata, keyName });
+                            // Apply change to current configuration
+                            prop.SetValue(config, result, null);
```

Comment:
Why not just call into Configuration.Current directly here rather than us passing it in as a 2nd param? It's the only one that's ever used.

Reply:
A change that I made, then I was let's try to not change too much code and keep the idea of ConfigurationExtra to be kind of methods to help.
At one point I did call the method something like `LoadRemoteConfigurationFromStream`.
I can make that change if you want ;-)

Reply:
I'd prefer to keep the original LoadFromIni overload and have a new function that overrides the local config with the server config.

Reply:
done :)